reframe/core/meta.py

```diff
@@ -79,12 +79,21 @@ def __init__(cls, name, bases, namespace, **kwargs):
                            f"you should use the pipeline hooks instead")
                     raise ReframeSyntaxError(msg)
 
+    def __call__(cls, *args, **kwargs):
+        obj = cls.__new__(cls, *args, **kwargs)
+
+        # Intercept constructor arguments
+        if '_rfm_use_params' in kwargs:
+            del kwargs['_rfm_use_params']
+
+        obj.__init__(*args, **kwargs)
+        return obj
```

Comment:
Nice trick! I haven't thought about it. I think you can write more simply though:

```suggestion
    def __call__(cls, *args, **kwargs):
        kwargs.pop('_rfm_use_params', None)
        return super().__call__(*args, **kwargs)
```

Reply:
Ok, now I see why you are doing as such. You want to call `__new__()` and `__init__()` with different keyword arguments. So ignore my comment here and the other below! Perhaps you need to comment this further so as to help others understand the trick here.

Reply:
I've still changed that `if '_rfm_use_params' in kwargs` by `kwargs.pop('_rfm_use_params', None)` since it reads much better. I've also added some comments to the `__call__` method to explain what it does.